airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -36,7 +36,13 @@ class ManifestDeclarativeSource(DeclarativeSource):
 
     VALID_TOP_LEVEL_FIELDS = {"check", "definitions", "schemas", "spec", "streams", "type", "version"}
 
-    def __init__(self, source_config: ConnectionDefinition, debug: bool = False, component_factory: ModelToComponentFactory = None):
+    def __init__(
+        self,
+        source_config: ConnectionDefinition,
+        debug: bool = False,
+        emit_connector_builder_messages: bool = False,
+        component_factory: ModelToComponentFactory = None,
```

Comment:
i thought about suggesting that we just get rid of this, but the legacy connector server still passes this in so it's probably easier to just leave it in the short term. But when we do clean out the old version, we should also do our due diligence to clean up some of these inputs if they're no longer necessary

Reply:
Good call. Should we leave a TODO? Or is there a ticket for deleting the legacy server that we can add this to in order to keep track of it?

Reply:
Getting rid of what exactly? 

the custom ModelToComponentFactory is still needed with the new code. I was thinking of moving the instantiation to a factory method so we don't need to manually set the `emit_connector_builder_messages` twice.

I'll add any clean up tasks [to this ticket's acceptance criteria](https://github.com/airbytehq/airbyte/issues/23127)

Reply:
An alternative would be to pass the `TestReadLimits` to `ManifestDeclarativeSource` and let it create the `ModelToComponentFactory()`. I'm also fine with that

Reply:
Sorry I should have been more clear, because we are moving `emit_connector_builder_messages` to be passed while initializing the `ManifestDeclarativeSource`, we no longer need to have `connector_builder/main.py` pass in its own ModelToComponentFactory and can just rely on the default. So long term I meant we can get rid of the parameter and instantiate the default ModelToComponent always.

Only reason we can't do it now is the legacy builder server is still passing in a factory if I recall